src/content/docs/en/extend/add-view-transitions.mdx

```diff
@@ -0,0 +1,351 @@
+
+    Add this new script to the `<ThemeIcon />` component, in addition to the script that controls the theme toggle.
+
+    ```astro title="src/components/ThemeIcon.astro" ins={3-9}
+    <script> ... <script>
+
+    <script>
+      document.addEventListener('astro:after-swap', () => {
+        localStorage.theme === 'dark' 
+        ? document.documentElement.classList.add("dark")
+        : document.documentElement.classList.add("light");
+      }); 
+    </script>
+    ```
+
+    Now, every page change **that uses the `<ViewTransitions />` router for client-side navigation** (and therefore access to the `astro:after-swap` event) will be able to detect `theme: dark` from the browser's `localStorage` and update the current page accordingly before the page is rendered for the viewer.
+
+### Customize Transition animations
+
+7. Change the default `fade` animation to a `slide` for the page title.
+
+    With view transitions enabled, you currently have a small fade in and out set for all page transition animations. Astro also provides a built-in `slide` animation. To change the type of animation for a single element, add the `transition:animation=""` directive.
+
+    For example, to make the page titles slide in instead of fade in, add `transition:animate="slide"` to the `<h1>` element in the BaseLayout:
+
+    ```astro title="src/layouts/BaseLayout.astro" ins='transition:animate="slide"'
+    ---
+    import Header from "../components/Header.astro";
+    import Footer from "../components/Footer.astro";
+    import "../styles/global.css";
+    import { ViewTransitions } from 'astro:transitions';
+    const { pageTitle } = Astro.props;
+    ---
+
+    <html lang="en">
+      <head>
+        <meta charset="utf-8" />
+        <link rel="icon" type="image/svg+xml" href="/favicon.svg" />
+        <meta name="viewport" content="width=device-width" />
+        <meta name="generator" content={Astro.generator} />
+        <title>{pageTitle}</title>
+        <ViewTransitions />
+      </head>
+      <body>
+        <Header />
+        <h1 transition:animate="slide">{pageTitle}</h1>
+        <slot />
+        <Footer />
+        <script>
+          import "../scripts/menu.js";
+        </script>
+      </body>
+    </html>
+    ```
+
+    In your browser preview, you will now see the page titles slide on to the screen, while other elements such as the body text continues to fade in and out. 
+
+    <Box icon="puzzle-piece">
+      ### Try it yourself - Make the navigation links slide in
+
+      Add an animation directive to make the `<div>` containing all the header links slide in on page navigation, following the [same steps as above](#customize-transition-animations).
```

Comment:
Specified `Navigation.astro` to guide people towards the component. Maybe the sentence can be reworked a little too :thinking: 
```suggestion
      Add an animation directive to make the `<div>` in `Navigation.astro` containing all the header links slide in on page navigation, following the [same steps as above](#customize-transition-animations).
```